### ISSUE TITLE
Large Commented SQL Causes Stack Overflow java.util.regex.Pattern, cannot upgrade beyond 5.0

### ISSUE THREAD

**Issue by qabulin** [dbeaver/dbeaver#4149]
I have not been able to upgrade beyond 5.0.6.  I tried the latest 5.1.6 and still get this error (see below) when I highlight a query that has a large commented junk in the middle. Usually its a chunk of 100+ lines of commented code in the middle of the entire highlighted block.

Steps to reproduce:
* connection is to Redshift, using driver version 42-1.2.15.1025
* select and highlight hundreds of lines of SQL code, perhaps multiple queries, with a large chunk in the middle commented out
* select "Execute SQL Statement"

I get a pop up saying there was an internal JDBC error, and the error below.  I've tried different versions of Redshift drivers with no success.  I tried changing the Xmx settings in Contents / Eclipse / dbeaver.ini, but this did not help.  If I delete the commented chunk of code in the middle, the error goes away, but this is not a feasible solution as I'm quickly testing code changes.

Any suggestions?

```
!ENTRY org.jkiss.dbeaver.model 4 0 2018-09-04 11:32:08.342
!MESSAGE SQL Error: Internal jdbc driver error
!SUBENTRY 1 org.jkiss.dbeaver.model 4 0 2018-09-04 11:32:08.342
!MESSAGE Internal jdbc driver error
!STACK 0
java.sql.SQLException: Internal jdbc driver error
        at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.handleExecuteError(JDBCStatementImpl.java:309)
        at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:343)
        at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:134)
        at org.jkiss.dbeaver.runtime.sql.SQLQueryJob.executeStatement(SQLQueryJob.java:445)
        at org.jkiss.dbeaver.runtime.sql.SQLQueryJob.lambda$0(SQLQueryJob.java:389)
        at org.jkiss.dbeaver.model.DBUtils.tryExecuteRecover(DBUtils.java:1559)
        at org.jkiss.dbeaver.runtime.sql.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:387)
        at org.jkiss.dbeaver.runtime.sql.SQLQueryJob.extractData(SQLQueryJob.java:819)
        at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.readData(SQLEditor.java:2346)
        at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:93)
        at org.jkiss.dbeaver.model.DBUtils.tryExecuteRecover(DBUtils.java:1559)
        at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:91)
        at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:95)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)
Caused by: java.lang.StackOverflowError
        at java.util.regex.Pattern$Loop.match(Pattern.java:4787)
        at java.util.regex.Pattern$GroupTail.match(Pattern.java:4719)
        at java.util.regex.Pattern$BranchConn.match(Pattern.java:4570)
        at java.util.regex.Pattern$CharProperty.match(Pattern.java:3779)
        at java.util.regex.Pattern$Branch.match(Pattern.java:4606)
        at java.util.regex.Pattern$GroupHead.match(Pattern.java:4660)
        at java.util.regex.Pattern$Loop.match(Pattern.java:4787)
        at java.util.regex.Pattern$GroupTail.match(Pattern.java:4719)
        at java.util.regex.Pattern$BranchConn.match(Pattern.java:4570)
        at java.util.regex.Pattern$CharProperty.match(Pattern.java:3779)
        at java.util.regex.Pattern$Branch.match(Pattern.java:4606)
        at java.util.regex.Pattern$GroupHead.match(Pattern.java:4660)
        at java.util.regex.Pattern$Loop.match(Pattern.java:4787)
        at java.util.regex.Pattern$GroupTail.match(Pattern.java:4719)
        at java.util.regex.Pattern$BranchConn.match(Pattern.java:4570)
        at java.util.regex.Pattern$CharProperty.match(Pattern.java:3779)
        at java.util.regex.Pattern$Branch.match(Pattern.java:4606)
        at java.util.regex.Pattern$GroupHead.match(Pattern.java:4660)
```

**Comment by serge-rider** [comment]
After 5.0.6 we've upgraded Redshift driver version to 1.2.15. This version contains a bug - for some reason it executes some regex matching with bad regex pattern (I've debugged this and see that it tries to match SQL query against pattern \0, i.e. zero character). This causes stack overflow error in case of long queries.

Workarounds: 
- downgrade to previous Redshift driver version (https://docs.aws.amazon.com/redshift/latest/mgmt/jdbc-previous-versions.html, I've just tested it and version 1.2.12 seems to be alright)
- Use PostgreSQL legacy driver instead of Redshift driver.

In any case it is not a DBeaver issue but AWS driver issue.
